ci/si_pipeline.sh

```diff
@@ -37,49 +38,80 @@ function create-junit-xml {
 
 function exit-with-cluster-launch-error {
     echo "$1"
-    create-junit-xml "dcos-launch" "cluster.create" "$1"
-    pipenv run dcos-launch -i "$INFO_PATH" delete
+    create-junit-xml "terraform" "apply" "$1"
+    if [ -f "$TERRAFORM_STATE" ]; then terraform destroy -auto-approve -state "$TERRAFORM_STATE"; fi
     "$ROOT_PATH/ci/dataDogClient.sc" "marathon.build.$JOB_NAME_SANITIZED.cluster_launch.failure" 1
     exit 0
 }
 
 function download-diagnostics-bundle {
-	BUNDLE_NAME="$(pipenv run dcos node diagnostics create all | grep -oE 'bundle-.*')"
+	BUNDLE_NAME="$(dcos node diagnostics create all | grep -oE 'bundle-.*')"
 	echo "Waiting for bundle ${BUNDLE_NAME} to be downloaded"
-	STATUS_OUTPUT="$(pipenv run dcos node diagnostics --status)"
+	STATUS_OUTPUT="$(dcos node diagnostics --status)"
 	while [[ $STATUS_OUTPUT =~ "is_running: True" ]]; do
 		echo "Diagnostics job still running, retrying in 5 seconds."
 		sleep 5
-		STATUS_OUTPUT="$(pipenv run dcos node diagnostics --status)"
+		STATUS_OUTPUT="$(dcos node diagnostics --status)"
 	done
-	pipenv run dcos node diagnostics download "${BUNDLE_NAME}" --location=./diagnostics.zip
+	dcos node diagnostics download "${BUNDLE_NAME}" --location=./diagnostics.zip
 }
 
 # Install dependencies and expose new PATH value.
 # shellcheck source=../../ci/si_install_deps.sh
 source "$ROOT_PATH/ci/si_install_deps.sh"
 
 # Launch cluster and run tests if launch was successful.
-SHAKEDOWN_SSH_KEY_FILE="$(pwd)/$DEPLOYMENT_NAME.pem"
-export SHAKEDOWN_SSH_KEY_FILE
-
 SHAKEDOWN_SSH_USER="centos"
 export SHAKEDOWN_SSH_USER
 
+# Configure cluster.
```

Comment:
There is [this wiki article](https://wiki.mesosphere.com/display/MARATHON/Scripting+Languages) about scripting languages (written by a certain K.J) which states:
```
When to Choose Bash
You can choose Bash if

the script is below 10 lines 
any engineering team member can grasp the semantics in a few minutes.
the script is easy to extent and modify.
```
I guess there was a time when this file actually met these criteria but this time is long gone now. I propose to rewrite this in Ammonite.